srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java

```diff
@@ -17,6 +33,146 @@
+                NumericalTerm numTerm = (NumericalTerm)modelTerm;
+                dispName = numTerm.getVariable().getDisplayName();                                     
+            }
+            else if( modelTerm instanceof DiscreteTerm )
+            {
+                DiscreteTerm discreteTerm = (DiscreteTerm)modelTerm;                                
+                MultiSelectOption opt = discreteTerm.getOption();
+                DiscreteVariable var = discreteTerm.getVariable();
+                
+                if( var instanceof DiscreteNumericalVariable ) 
+                {
+                    type = "Discrete Numeric";
+                    dispName = var.getDisplayName()+ " = " + opt.getValue();
+                }
+                else if( var instanceof MultiSelectVariable ) 
+                {
+                    type = "Discrete";
+
+                    MultiSelectVariable msv = (MultiSelectVariable)var;
+                    msv.getOptions();
+                    dispName = var.getDisplayName()+ " = " + opt.getValue() ;
+                }
+            }            
+            else if( modelTerm instanceof DerivedTerm )
+            {
+                type = "Rule";
+                dispName = "Rule: "+((DerivedTerm)modelTerm).getRule().getId();
+            }
+            else
+            {
+                // or throw an error?
+                dispName = "Unrecognized Term: "+modelTerm.getClass().getName();    
+                type = "";
+            }
+            
+            // Use new Type column to display the Type.
+            // Can still do this if that's what they want.
+            // dispName = dispName + " ("+type+")";
+            
+            return new ModelTermSummary( modelTerm, dispName, type );
+        }
+
+        // a string used to make sorting the terms easier. 
+        // ConstantTerms first, then Rules and then by display name 
+        // with discrete options ordered by index 
+        private String getSortString() 
+        {
+            if( targetTerm instanceof ConstantTerm ) 
+            {
+                return "AAAA";
+            }
+            else if( targetTerm instanceof DerivedTerm )
+            {
+                return "AAAAA"+Float.toString( ((DerivedTerm)targetTerm).getRule().getId()*1f/1000f ); // 7 before 11
+            }
+            // order DiscreteTerms according to their options
+            //
+            else if( targetTerm instanceof DiscreteTerm )
+            {
+                DiscreteTerm dTerm = (DiscreteTerm)targetTerm;
+                return dTerm.getVariable().getDisplayName().toUpperCase() + dTerm.getOptionIndex();
+            }
+            else
+            {
+                return displayName.toUpperCase();
+            }
+        }
+        
+        // sort by the sortString
+        @Override
+        public int compareTo( ModelTermSummary o ) 
+        {
+            return getSortString().compareTo( o.getSortString() );
```

Comment:
I don't think having a "sort string" is a good idea here since you can use the compareTo() method to sort by type instead. The getSortString() method should be removed in favor of a better constructed compareTo() method.